shared/src/test/scala/scorex/crypto/hash/HashTest.scala

```diff
@@ -56,17 +56,18 @@ trait HashTest extends AnyPropSpec
       }
     }
 
-
-    property(s"${hash.getClass.getSimpleName} should return correct Tag") {
-      forAll { (string: String, bytes: Array[Byte]) =>
-        val digest = hash(string)
-        digest.isInstanceOf[D] shouldBe true
-        if (digest.isInstanceOf[Digest32]) {
-          hash.DigestSize shouldBe 32
-        } else if (digest.isInstanceOf[Digest64]) {
-          hash.DigestSize shouldBe 64
-        }
-      }
-    }
+    // TODO refactor: reimplement Digest classes to pass runtime check or remove this test
+    // test incorrect in Scala2 last check not reached
+//    property(s"${hash.getClass.getSimpleName} should return correct Tag") {
+//      forAll { (string: String, bytes: Array[Byte]) =>
+//        val digest = hash(string)
+//        digest.isInstanceOf[D] shouldBe true
```

Comment:
Maybe leave it intact until here.

Reply:
> What's the point?

Just the general rule to leave (test) code intact.

> It double checks the compiler since `def hash(input: String): D`

If you are 100% sure that the test is redundant under any and all circumstances, then you can remove it of course.